### PR TITLE
[5.8] Use Name of Factory as Model

### PR DESCRIPTION
**Pull request by simoebenhida**
Use the name of factory as the model if --model is not specified

```php artisan make:factory UserFactory```

the Model will be `User`

### CODE REVIEW

**Comment by taylorotwell** [comment]
What if the name of the model genuinely has "Factory" inside of it?

File: src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php [hunk 1]
```diff
@@ -46,11 +47,18 @@ protected function getStub()
...
     {
-        $namespaceModel = $this->option('model')
-                        ? $this->qualifyClass($this->option('model'))
-                        : trim($this->rootNamespace(), '\\').'\\Model';
+        $model = class_basename($this->option('model') ?? $name);
 
-        $model = class_basename($namespaceModel);
+        foreach (['factory', 'Factory'] as $name) {
+            if (Str::endsWith($model, $name) && strpos($model, $name)) {
+                $model = substr($model, 0, 4);
```
Comment (taylorotwell) [inline src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php]: What is this doing? Why are you slicing off a hard-coded 4 characters?

Comment (simoebenhida) [reply src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php]: Oh sorry about that . Was thinking of slicing the last 7 characters of `Factory` but didn't notice the `4` it is fixed now on the last commit.

File: tests/Console/Factory/FactoryMakeCommandTest.php [hunk 2]
```diff
@@ -41,6 +41,21 @@ public function it_is_giving_a_default_model_name_from_the_factory_name()
         $this->assertTrue((bool) strpos($slug, 'App\\Models\\User'));
     }
 
+    /** @test */
+    public function it_is_giving_a_default_model_name_from_the_factory_name_long_carachter()
```
Comment (taylorotwell) [inline tests/Console/Factory/FactoryMakeCommandTest.php]: carachter?

File: tests/Console/Factory/FactoryMakeCommandTest.php [hunk 3]
```diff
@@ -42,7 +42,7 @@ public function it_is_giving_a_default_model_name_from_the_factory_name()
     }
 
     /** @test */
-    public function it_is_giving_a_default_model_name_from_the_factory_name_long_carachter()
+    public function it_is_giving_a_default_model_name_from_the_factory_name_long_character()
```
Comment (simoebenhida) [inline tests/Console/Factory/FactoryMakeCommandTest.php]: @taylorotwell it is fixed on the second commit :D

File: src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php (hunk 1 above)
Comment (martinbean) [reply src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php]: @simoebenhida Might be nice to tuck it behind a nicely-named method, i.e.

```php
$model = $this->removeFactorySuffix($model);
```

File: src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php [hunk 4]
```diff
@@ -46,11 +47,18 @@ protected function getStub()
...
     protected function buildClass($name)
     {
-        $namespaceModel = $this->option('model')
-                        ? $this->qualifyClass($this->option('model'))
-                        : trim($this->rootNamespace(), '\\').'\\Model';
+        $model = class_basename($this->option('model') ?? $name);
 
-        $model = class_basename($namespaceModel);
+        foreach (['factory', 'Factory'] as $name) {
+            if (Str::endsWith($model, $name) && strpos($model, $name)) {
```
Comment (taylorotwell) [inline src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php]: What is the point of the strpos check here?

**Comment by taylorotwell** [comment]
Holding off on this for now.

**Pull request closed without merging**